qaframework-bdd-tests/src/test/resources/features/refapp-2.x/10-allergy/allergies.feature

```diff
@@ -5,13 +5,15 @@ Feature: Allergies Management
     Then the system loads Allergies page
 
   @selenium
-  @dashboard
-  Scenario: Add No known allergies
+  @allergies
+  Scenario: Allergies management
+  	# User story: Add No known allergies
     And a user clicks No Known Allergy button
     Then the system add no known allergies into the allergies table
-
-  @selenium
-  @dashboard
-  Scenario: Remove No known allergies
+    # User story: Remove No known allergies
     And a user clicks Remove No Known Allergy icon
     Then the system displays unknown in the allergies table
+    # User story: Add, Edit and Delete known allergies
+    When a user adds a known allergy into the system
+    Then a user edits a known allergy
+    And a user deletes a known allergy
```

Comment:
I thought Add allergy, Edit allergy and Deleting allergy each would be a separate scenario,  @Irenyak1 what do you think?

Reply:
@kdaud  initially I thought so but when @sherrif10 advised me to follow the approach of the[ form test](https://github.com/openmrs/openmrs-contrib-qaframework/blob/0d34bada84b09c2d83cc40043a8a3bb392af20b4/qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/FormSteps.java#L71) I thought I needed to bundle the steps up. What are your thoughts on this cc @kdaud , @sherrif10 .

Reply:
> I thought Add allergy, Edit allergy and Deleting allergy each would be a separate scenario, @Irenyak1 what do you think?

@kdaud  could they be just separate user stories and not separate scenario since every scenario will be dealing with a different patient yet I would want to the entire test to use the same patient. What are your suggestions?

Reply:
BDD framework uses cucumber feature files to hold use stories. These stories clearly indicates what the user does on a functionality. 

> I thought I needed to bundle the steps up

By bundling the steps up would mean some steps on a scenario are left out which is not the ideal for BDD framework. For example when a user is adding an allergy, the anatomy of the scenario would be something like
```js
When a user clicks on Allergies link from Patient dashboard page
Then the system loads Allergies board page
And a user selects an allergy
And a user clicks on save allergy button
Then the system adds known allergies into the allergies table
```

Reply:
> @kdaud could they be just separate user stories and not separate scenario since every scenario will be dealing with a different patient yet I would want to the entire test to use the same patient

Populatingall the user stories on the same patient is the better approach and the ideal would be for each user story the test returns on the Patient Dashbaord where the next user story should pick up its run.

Reply:
> BDD framework uses cucumber feature files to hold use stories. These stories clearly indicates what the user does on a functionality.
> 
> > I thought I needed to bundle the steps up
> 
> By bundling the steps up would mean some steps on a scenario are left out which is not the ideal for BDD framework. For example when a user is adding an allergy, the anatomy of the scenario would be something like
> 
> ```js
> When a user clicks on Allergies link from Patient dashboard page
> Then the system loads Allergies board page
> And a user selects an allergy
> And a user clicks on save allergy button
> Then the system adds known allergies into the allergies table
> ```
@kdaud  I fully understand your point. My request is that we need to harmonize and have one route to follow. The form test like [in the above comment ](https://github.com/openmrs/openmrs-contrib-qaframework/pull/298#discussion_r776857481) is a bit different which may confuse a new comer, in case they looked through our code. @sherrif10  what was the reason that you bundled up the steps in the form test? Was it a special case?

Reply:
> Populating all the user stories on the same patient is the better approach and the ideal would be for each user story the test returns on the Patient Dashbaord where the next user story should pick up its run.

@kdaud following this comment, it means that if the test has added a known allergy then it has to go back to the dashboard and the next user story which is editing the allergy will again start from the dashboard page then to the allergies board page and finally go to the edit allergy page. The following user stories will also do the same?

Reply:
@Irenyak1 what is your plan regarding implementing the test?

Reply:
@kdaud My plan was to let every step pick up from the allergies board page. However, if returning to the patient 
dashboard page is the right thing then I can still implement that.

Reply:
> My plan was to let every step pick up from the allergies board page

sounds a good plan.